### PR TITLE
feat: add fromiter to numpy frontend

### LINKED ISSUE #22814: fromiter

### PR DESCRIPTION
**Pull request by nwanne56**
# PR Description 

## Related Issue 

Close #22814

## Checklist 

- [X] Did you add a function?
- [X] Did you add the tests?
- [X] Did you follow the steps we provided?

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_shape_or_value.py [hunk 1]
```diff
@@ -234,6 +234,33 @@ def test_numpy_fromfunction(
     )
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_shape_or_value.py:236]: Please add:
`# fromiter` here

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_shape_or_value.py [hunk 2]
```diff
@@ -234,6 +234,33 @@ def test_numpy_fromfunction(
...
+    iterable,
+    dtype,
+    frontend,
+    backend_fw,
+    test_flags,
+    fn_tree,
+    on_device,
+):
+    helpers.test_frontend_function(
+        iterable=iterable,
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_shape_or_value.py:252]: the "iterable" here should be the same as your function's first arguement "iter"

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello, your tests are currently all failing for the same error:
`E   TypeError: fromiter() got an unexpected keyword argument 'iterable'`
Please read the comments below

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_shape_or_value.py (hunk 2 above)
Comment (nwanne56) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_from_shape_or_value.py:252]: Sorry about that, fixed.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Please update your test by using helpers.dtype_and_values to generate the iter argument as your current method will cause the following error when it generate a massive value:
E   OverflowError: Python int too large to convert to C long
Read this(https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html) for more details.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Please try to copy the changes you have made to a new branch in your repo as it is not a good practice to merge from your main to ivy main. In addition, there are some major changes applied to the code base recently, so take this as a chance to update your repo as well. Thanks.

**Pull request closed without merging**